### PR TITLE
Refactor - Runner

### PR DESCRIPTION
**Pull request by sivkovic**
## Changelog
1. Refactoring runner
2. Detecting file type based on extension
3. Migrating file paths to bid on db_value
4. Tests

### CODE REVIEW

File: file_system/models.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import uuid
 from enum import IntEnum
 from django.db import models
-from django.contrib.postgres.fields import JSONField
+from django.contrib.postgres.fields import JSONField, ArrayField
```
Comment (stevekm) [inline file_system/models.py]: did you plan to use `ArrayField` somewhere in here?

File: requirements-dev.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+dictdiffer==0.8.1
+mock==3.0.5
```
Comment (stevekm) [inline requirements-dev.txt]: can we just put this in the common `requirements.txt`?

**Comment by stevekm** [comment]
I still get the LDAP errors when running test;

```
$ python manage.py test
...
.Caught LDAPError while authenticating username: LDAPError(0, 'Error')
.Caught LDAPError while authenticating username: LDAPError(0, 'Error')
METADATA SAVE
...
```

did this branch not already inherit the fix for this? Or has that not been fixed yet?

**Comment by stevekm** [comment]
tests passed for me; 

```
$ make test
unset TMPDIR ; \
	python manage.py test \
	runner.tests.operator.roslin_operator.test_pair_request \
	runner.tests.operator.roslin_operator.test_make_sample \
	runner.tests.operator.roslin_operator.test_construct_roslin_pair \
	runner.tests.run.test_processors \
	runner.tests.run.test_run
.................
----------------------------------------------------------------------
Ran 25 tests in 2.089s

OK
```

```
$ make test-lims
python manage.py test \
	beagle_etl.tests.jobs.test_lims_etl_jobs
.....
----------------------------------------------------------------------
Ran 8 tests in 13.810s
```

I updated the Makefile to include the extras from this PR

File: file_system/models.py (hunk 1 above)
Comment (sivkovic) [reply file_system/models.py]: I was planing and then forgot to remove import when I decided for different implementation. Will remove it. Thanks

**Pull request merged**